pkgs/development/tools/locate-dominating-file/default.nix

```
@@ -0,0 +1,48 @@
+{
+  bats,
+  bash,
+  lib,
+  stdenv,
+  getopt,
+  fetchFromGitHub,
+}: let
+  batsPkg = bats.withLibraries (p: [p.bats-support p.bats-assert]);
+  version = "v0.0.1";
+in
+  stdenv.mkDerivation {
+    name = "locate-dominating-file";
+    inherit version;
+    src = fetchFromGitHub {
+      owner  = "roman";
+      repo   = "locate-dominating-file";
+      rev    = version;
+      sha256 = "sha256-gwh6fAw7BV7VFIkQN02QIhK47uxpYheMk64UeLyp2IY=";
+    };
+
+    doCheck = true;
+
+    postPatch = ''
+      for file in $(find src tests -type f); do
+        substituteInPlace "$file" --replace "/bin/bash" "${bash}/bin/bash"
+        substituteInPlace "$file" --replace "#!/usr/bin/env bash" "#!${bash}/bin/bash"
+      done
+    '';
+
```

As I have understood, bats is only used to test. So, add this:

```suggestion

  checkInputs = [
    (bats.withLibraries (p: [ p.bats-support p.bats-assert ]))
  ];

```